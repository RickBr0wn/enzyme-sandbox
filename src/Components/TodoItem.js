import React from 'react'
import PropTypes from 'prop-types'

const TodoItem = ({ todo }) => {
  return <li className='todo-item'>{todo}</li>
}

TodoItem.propTypes = {
  todo: PropTypes.string
}

export default TodoItem
