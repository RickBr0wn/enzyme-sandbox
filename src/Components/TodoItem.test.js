import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import TodoItem from './TodoItem'

configure({ adapter: new Adapter() })

describe('<TodoItem />', () => {
  it('should render correctly', () => {
    const todoItemComponent = renderer.create(<TodoItem />)
    const tree = todoItemComponent.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should return an `li` tag', () => {
    const todoItemComponent = shallow(<TodoItem />)
    const liTag = todoItemComponent.find('li')
    expect(liTag).toBeTruthy()
  })

  it('should return only 1 `li` tag', () => {
    const todoItemComponent = shallow(<TodoItem todo={'Testing'} />)
    expect(todoItemComponent.find('li')).toHaveLength(1)
  })

  it('should correctly display the props', () => {
    const todoItemComponent = shallow(<TodoItem todo={'Testing'} />)
    const li = <li className='todo-item'>Testing</li>
    expect(todoItemComponent.contains(li)).toBe(true)
  })
})
