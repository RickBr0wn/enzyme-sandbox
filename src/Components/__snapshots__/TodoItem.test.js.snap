// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> should render correctly 1`] = `
<li
  className="todo-item"
/>
`;
